### ISSUE TITLE
"Undefined offset: 0" when multiple repeaters with similar names stored in the viewBag

### ISSUE THREAD

**Issue by charisketikidis** [octobercms/october#5591]
When multiple repeaters with similar names stored in theme form fields yaml file:
`""Undefined offset: 0" on line 425 of /.../..../modules/backend/formwidgets/Repeater.php" on line 567 of /.../..../modules/backend/classes/Controller.php`

To reproduce:
Call those repeater fields in your theme settings: 

```
tabs:
    fields:
        settings:
            tab: Settings
            label: Settings Items
            prompt: Add item
            type: repeater
            groups:
                settings_title:
                    name: Title
                    fields:
                        settings_item_title:
                            label: Title
                            type: text
                settings_desc:
                    name: Description
                    fields:
                        settings_item_desc:
                            label: Description
                            type: text
        settings_second:
            tab: Settings
            label: Settings Second Items
            prompt: Add item
            type: repeater
            groups:
                settings_second_title:
                    name: Title
                    fields:
                        settings_second_item_title:
                            label: Title
                            type: text
                settings_second_desc:
                    name: Description
                    fields:
                        settings_second_item_desc:
                            label: Description
                            type: text
```

Then on the backend if you try to create one field from the first repeater and one field from the second you will see error:
`""Undefined offset: 0" on line 425 of /.../..../modules/backend/formwidgets/Repeater.php" on line 567 of /.../..../modules/backend/classes/Controller.php`

![Screenshot_6](https://user-images.githubusercontent.com/6757833/116676989-575d4280-a9b0-11eb-8343-d4b720321e0d.png)

If you change the name of the "settings_second" repeater to "test_second"
 everything is working ok.

This error is similar to this one, if that helps:
https://github.com/octobercms/october/issues/4808

**Comment by daftspunk** [comment]
This fix has been reviewed and implemented in `develop`, please test and confirm

**Comment by daftspunk** [comment]
Should be fixed in v2.0.12

Thanks!
